bootstrap.sh

```diff
@@ -0,0 +1,81 @@
+# #!/bin/bash
+sudo rm -rf /etc/blocktogether
+sudo mkdir /etc/blocktogether
+cp -r /vagrant/config/* /etc/blocktogether/
+mv /etc/blocktogether/development.json /etc/blocktogether/config.json
+
+cd /vagrant
+
+# Set up a production instance of Block Together. Assumes no existing root MySQL
+# password (as is true of a freshly installed mysql-server).
+# Safe to run multiple times.
+#
+DB_ROOT_PASS=${DB_ROOT_PASS:-$(openssl rand -hex 20)}
+if [ ! -f /usr/sbin/mysqld ] ; then
+  sudo debconf-set-selections <<<"mysql-server mysql-server/root_password password $DB_ROOT_PASS"
+  sudo debconf-set-selections <<<"mysql-server mysql-server/root_password_again password $DB_ROOT_PASS"
+fi
+
+DB_PASS=$(openssl rand -hex 20)
+sudo apt-get update
+sudo apt-get install -y mysql-client mysql-server git nginx gnupg curl build-essential vim
+
+SEQUELIZE_CONFIG=/etc/blocktogether/sequelize.json
+if grep -q __PASSWORD__ $SEQUELIZE_CONFIG ; then
+  sed -i s/__PASSWORD__/$DB_PASS/g $SEQUELIZE_CONFIG
+  mysql -u root --password="$DB_ROOT_PASS" <<EOSQL
+    CREATE DATABASE IF NOT EXISTS blocktogether;
+    GRANT ALL PRIVILEGES ON blocktogether.* to 'blocktogether'@'localhost' IDENTIFIED BY "${DB_PASS}";
+EOSQL
+fi
+
+COOKIE_SECRET=$(openssl rand -hex 20)
+sed -i s/__COOKIE_SECRET__/$COOKIE_SECRET/g /etc/blocktogether/config.json
+
+CONF=/etc/blocktogether
+if [ ! -f ${CONF}/rpc.key ] ; then
+  openssl req -new -newkey rsa:2048 -nodes -days 10000 -x509 \
+    -keyout ${CONF}/rpc.key -out ${CONF}/rpc.crt \
+    -subj /CN=blocktogether-rpc
+  chmod 0600 ${CONF}/rpc.key
+fi
+
+if ! crontab -l >/dev/null; then
+  crontab - <<-EOCRON
+    PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games
+    MAILTO=ubuntu
+    23 10 * * * bash /data/blocktogether/current/util/cron.sh
+EOCRON
+fi
+
+su - vagrant -c'source /vagrant/install_nvm.sh'
```

Comment:
This script is meant to be run as a non-privileged user, and occasionally escalate using sudo. The nice thing about that is we don't have to hardcode the username of the non-privileged user account, which is 'vagrant' on vagrant or 'ubuntu' on AWS.